C++ build failure on macOS Ventura: deprecated sprintf

```
src/IceUtil/OutputUtil.cpp:38:5: error: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Werror,-Wdeprecated-declarations]
    sprintf(buf, "%ld", val); // Avoids a format warning from GCC.
    ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdio.h:188:1: note: 'sprintf' has been explicitly marked deprecated here
__deprecated_msg("This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead.")
^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/sys/cdefs.h:214:48: note: expanded from macro '__deprecated_msg'
        #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
                                                      ^
1 error generated.
make: *** [src/IceUtil/build/macosx/static/OutputUtil.o] Error 1
make: *** Waiting for unfinished jobs....
```

Similar issue with the php mapping:
```
src/php/Util.cpp:785:5: error:
'vsprintf' is deprecated: This function is provided for compatibility reasons only. Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use vsnprintf(3) ins
tead. [-Werror, -Wdeprecated-declarations]
vsprintf (msg, fmt, args);
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdio.h:207:1:note:
'vsprintf' has been explicitlv marked deprecated here
deprecated_msg ("This function is provided for compatibility reasons only.
Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use vsnprintf(3) instead.")
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/sys/cdefs.h:214:48:note:expandedfrommacro
#define
deprecated msg(_msg)
attribute
_(l
_deprecated
_( msg)))
deprecated msg'
src/php/Util.cpp:804:5:error:
'vsprintf" is deprecated: This function is provided for compatibility reasons only. Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use vsnprintf(3) ins
ted. [-Werror,-Wdeprecated-declarations]
vsprintf (msg, fmt, args);
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdio.h:207:1:note:
'vsprintf' has been explicitlv marked deprecated here
deprecated msg("This function is provided for compatibility reasons only.
Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use vsnprintf(3) instead.")
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/sys/cdefs.h:214:48:note:expandedfrommacro
#define
deprecated msg( msg)
attribute ((
deprecated (msg)))
deprecated msg'
2 errors generated.
make: *** [src/php/build/macosx/shared/pic/Util.o]Error1
```

--- a/cpp/src/IceUtil/OutputUtil.cpp
+++ b/cpp/src/IceUtil/OutputUtil.cpp
@@ -36,11 +36,7 @@ IceUtilInternal::int64ToString(Int64 val)
     sprintf_s(buf, sizeof(buf), "%I64d", val);
 #elif defined(__APPLE__)
     // sprintf is deprecated with macOS Ventura
-#   if defined(ICE_64)
     snprintf(buf, sizeof(buf), "%ld", val); // Avoids a format warning from GCC.
-#   else
-    snprintf(buf, sizeof(buf), "%lld", val);
-#   endif
 #else
 #   if defined(ICE_64)
     sprintf(buf, "%ld", val); // Avoids a format warning from GCC.

--- a/cpp/src/IceUtil/OutputUtil.cpp
+++ b/cpp/src/IceUtil/OutputUtil.cpp
@@ -34,10 +34,19 @@ IceUtilInternal::int64ToString(Int64 val)
 
 #if defined(_WIN32)
     sprintf_s(buf, sizeof(buf), "%I64d", val);
-#elif defined(ICE_64)
+#elif defined(__APPLE__)
+    // sprintf is deprecated with macOS Ventura
+#   if defined(ICE_64)
     snprintf(buf, sizeof(buf), "%ld", val); // Avoids a format warning from GCC.
-#else
+#   else
     snprintf(buf, sizeof(buf), "%lld", val);
+#   endif
+#else
+#   if defined(ICE_64)
+    sprintf(buf, "%ld", val); // Avoids a format warning from GCC.
+#   else
+    sprintf(buf, "%lld", val);
+#   endif
 #endif
     return string(buf);
 }

--- a/cpp/src/IceUtil/OutputUtil.cpp
+++ b/cpp/src/IceUtil/OutputUtil.cpp
@@ -35,9 +35,9 @@ IceUtilInternal::int64ToString(Int64 val)
 #if defined(_WIN32)
     sprintf_s(buf, sizeof(buf), "%I64d", val);
 #elif defined(ICE_64)
-    sprintf(buf, "%ld", val); // Avoids a format warning from GCC.
+    snprintf(buf, sizeof(buf), "%ld", val); // Avoids a format warning from GCC.
 #else
-    sprintf(buf, "%lld", val);
+    snprintf(buf, sizeof(buf), "%lld", val);
 #endif
     return string(buf);
 }

--- a/php/src/php/Util.cpp
+++ b/php/src/php/Util.cpp
@@ -782,7 +782,7 @@ IcePHP::runtimeError(const char* fmt, ...)
 #if defined(_MSC_VER)
     vsprintf_s(msg, fmt, args);
 #else
-    vsprintf(msg, fmt, args);
+    vsnprintf(msg, sizeof(msg), fmt, args);
 #endif
 
     va_end(args);
@@ -801,7 +801,7 @@ IcePHP::invalidArgument(const char* fmt, ...)
 #if defined(_MSC_VER)
     vsprintf_s(msg, fmt, args);
 #else
-    vsprintf(msg, fmt, args);
+    vsnprintf(msg, sizeof(msg), fmt, args);
 #endif
 
     va_end(args);

--- a/php/src/php/Util.cpp
+++ b/php/src/php/Util.cpp
@@ -781,8 +781,11 @@ IcePHP::runtimeError(const char* fmt, ...)
 
 #if defined(_MSC_VER)
     vsprintf_s(msg, fmt, args);
-#else
+#elif defined(__APPLE__)
+    // vsprintf is deprecated with macOS Ventura
     vsnprintf(msg, sizeof(msg), fmt, args);
+#else
+    vsprintf(msg, fmt, args);
 #endif
 
     va_end(args);
@@ -800,8 +803,11 @@ IcePHP::invalidArgument(const char* fmt, ...)
 
 #if defined(_MSC_VER)
     vsprintf_s(msg, fmt, args);
-#else
+#elif defined(__APPLE__)
+    // vsprintf is deprecated with macOS Ventura
     vsnprintf(msg, sizeof(msg), fmt, args);
+#else
+    vsprintf(msg, fmt, args);
 #endif
 
     va_end(args);

